qiling/arch/arch.py

```diff
@@ -3,30 +3,44 @@
 # Cross Platform and Multi Architecture Advanced Binary Emulation Framework
 #
 
-from abc import ABC
-from typing import Optional
+from abc import abstractmethod
 
 from unicorn import Uc
 from unicorn.unicorn import UcContext
 from capstone import Cs
 from keystone import Ks
 
 from qiling import Qiling
+from qiling.const import QL_ENDIAN
+from .register import QlRegisterManager
 from .utils import QlArchUtils
 
-class QlArch(ABC):
+class QlArch:
+    bits: int
+
     def __init__(self, ql: Qiling):
         self.ql = ql
         self.utils = QlArchUtils(ql)
 
-        self._disasm: Optional[Cs] = None
-        self._asm: Optional[Ks] = None
+    @property
+    @abstractmethod
+    def uc(self) -> Uc:
+        """Get unicorn instance bound to arch.
+        """
+
+        pass
```

Comment:
I suggest raising a NotImplementedError here. Same below.

Reply:
The `@abstractmethod` decorator does that.